### DISCUSSION TITLE
CorePlot does not compile

### DISCUSSION THREAD

**Discussion by o-bhave** [core-plot/core-plot#475] [General]
Hi,

I look latest from master and the corePlot framework does not compile for me with lot of C99 related errors. 

Am I missing anything?

Omkar Bhave

**Comment by eskroch** [comment]
Use the code on the `release-2.4` branch.

**Reply by o-bhave** [reply]
release-2.4 does not compile either. It has lot of compilation errors. Please fix them

**Reply by o-bhave** [reply]
We really like CorePlot and we want to use it for our tools and analytics but these errors are preventing us from using it and we might have to consider to move to other libraries unless there is resolution.

**Reply by Dirk-** [reply]
@o-bhave You should be much more polite. Maintainers of useful and free libraries like CorePlot do not owe you anything.

**Reply by o-bhave** [reply]
I agree please see my comment below.

**Reply by o-bhave** [reply]
Dirk I must say that Core Plot is top class graphs library and people really look forward and rely on this library. That brings lot of onus on maintainers of it to keep it in good shape all the time in order to not only promote but keep it on the top. Think from users point. If they cannot compile it then they will certainly look for options and believe me this is a competition world and there are other options available. Since I have personal inclination towards core plots I felt an urge to bring this issue.

**Reply by Dirk-** [reply]
@o-bhave Thanks for reacting this way. That was a cool move.

**Comment by eskroch** [comment]
What version of Xcode are you using? i'm not seeing any errors with Xcode 14.2. Can you give some examples of the types of errors you're seeing?

**Comment by o-bhave** [comment]
I tried to compile it on Xcode 14.3 Beta2. I get following errors

/core-plot/framework/Source/_CPTMaskLayer.m:36:15 Mixing declarations and code is incompatible with standards before C99

Command PhaseScriptExecution failed with a nonzero exit code

I get lot other compilation errors as well. 

It is very hard to use core plots library and it is not user friendly. Earlier it used to be very easy to compile and use.

**Reply by Dirk-** [reply]
So you used CorePlot before and it worked. Then you installed a **beta** version of Xcode and it didn't work any more. This is not "not user friendly" by CorePlot.

It would have been nice if you started this thread like "Hey, I tried CorePlot on the latest beta version of Xcode and got lots of compiling errors. You might want to take a look at this."

**Reply by o-bhave** [reply]
I agree with you Dirk. Sorry for my impatience. This is a crunch time and we had certain reliance on this library and since latest builds are blocking our deliverables it was a natural reaction. I hope you understand.

**Comment by eskroch** [comment]
I just downloaded Xcode 14.3 beta 3 and I see what you mean. There are a bunch of new recommended project setting updates that I need to go through, too.

I don't understand where the C99 error is coming from. We've had the C99 compiler dialect selected in the project settings for years and never had the compiler complain.

I'm also seeing some warnings about quoted includes. See issue #433 for suggested workarounds. These errors were suppressed when compiling Core Plot in Xcode 14.2.

**Comment by eskroch** [comment]
This should be fixed on the `release-2.4` branch. Let me know if anyone is still seeing this issue with the latest code.

**Reply by o-bhave** [reply]
this is still needed. Please fix this.

**Reply by eskroch** [reply]
I've fixed all the compilation errors that I've come across. Do a clean build and post more detail about the error(s) that you're seeing to help us troubleshoot.

**Comment by eskroch** [comment]
I believe the build problems have been resolved on the `release-2.4` branch. Please let me know if you're still seeing any issues.
